pyramid_swagger/ingest.py

```diff
@@ -62,27 +82,76 @@ def _load_resource_listing(resource_listing):
         )
 
 
-def compile_swagger_schema(schema_dir):
+def generate_resource_listing(schema_dir, listing_base):
+    if 'apis' in listing_base:
+        raise ResourceListingGenerationError(
+            "{0}/{1} has an `apis` listing. Generating a listing would "
+            "override this listing.".format(schema_dir, API_DOCS_FILENAME))
+
+    return dict(
+        listing_base,
+        apis=[{'path': path} for path in find_resource_paths(schema_dir)]
+    )
+
+
+def get_resource_listing(schema_dir, should_generate_resource_listing):
+    """Return the resource listing document.
+
+    :param schema_dir: the directory which contains swagger spec files
+    :type  schema_dir: string
+    :param should_generate_resource_listing: when True a resource listing will
+        be generated from the list of *.json files in the schema_dir. Otherwise
+        return the contents of the resource listing file
+    :type should_enerate_resource_listing: boolean
+    :returns: the contents of a resource listing document
+    """
+    listing_filename = os.path.join(schema_dir, API_DOCS_FILENAME)
+    resource_listing = _load_resource_listing(listing_filename)
```

Comment:
Oh tricky. Yeah I'd express that more clearly in the configuration docs. I'm down for subsections devoted to explaining particularly non-intuitive or complex options. Just add a new one for your new feature and compare the before/after api_docs.json?

As for naming, I don't mind the current one. Maybe ``complete_resource_listing`` if we want to be explicit about the fact that we aren't generating from scratch?